File: core/src/main/java/io/github/fourlastor/game/di/modules/GameModule.java
package io.github.fourlastor.game.di.modules;

import com.badlogic.gdx.ai.msg.MessageDispatcher;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import dagger.Module;
import dagger.Provides;
import javax.inject.Singleton;
import squidpony.squidmath.IRNG;
import squidpony.squidmath.SilkRNG;

@Module
public class GameModule {

    @Provides
    @Singleton
    public IRNG random() {
        return new SilkRNG();
    }

    @Provides
    @Singleton
    public MessageDispatcher messageDispatcher() {
        return new MessageDispatcher();
    }

    @Provides
    @Singleton
    public SpriteBatch batch() {
        return new SpriteBatch();
    }
}

File: core/src/main/java/io/github/fourlastor/game/di/modules/GameModule.kt
package io.github.fourlastor.game.di.modules

import com.badlogic.gdx.ai.msg.MessageDispatcher
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import dagger.Module
import dagger.Provides
import squidpony.squidmath.IRNG
import squidpony.squidmath.SilkRNG
import javax.inject.Singleton

@Module
class GameModule {
    @Provides
    @Singleton
    fun random(): IRNG = SilkRNG()

    @Provides
    @Singleton
    fun messageDispatcher(): MessageDispatcher = MessageDispatcher()

    @Provides
    @Singleton
    fun batch(): SpriteBatch = SpriteBatch()
}
